Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabSystemComponent.cpp

```diff
@@ -92,7 +92,25 @@ namespace AzToolsFramework
 
         AZStd::unique_ptr<Instance> PrefabSystemComponent::CreatePrefab(
             const AZStd::vector<AZ::Entity*>& entities, AZStd::vector<AZStd::unique_ptr<Instance>>&& instancesToConsume,
-            AZ::IO::PathView filePath, AZStd::unique_ptr<AZ::Entity> containerEntity, bool shouldCreateLinks)
+            AZ::IO::PathView filePath, AZStd::unique_ptr<AZ::Entity> containerEntity)
+        {
+            AZStd::unique_ptr<Instance> newInstance = AZStd::make_unique<Instance>(AZStd::move(containerEntity));
+            return CreatePrefab(entities, AZStd::move(instancesToConsume), filePath, AZStd::move(newInstance), true);
+        }
+
+        AZStd::unique_ptr<Instance> PrefabSystemComponent::CreatePrefabUnderParent(
+            const AZStd::vector<AZ::Entity*>& entities, AZStd::vector<AZStd::unique_ptr<Instance>>&& instancesToConsume,
+            AZ::IO::PathView filePath, InstanceOptionalReference parent)
```

Comment:
The way we are hardcoding the 'shouldCreateLinks' for different flavors of 'CreatePrefab' is not a good idea IMO because we'd be saying to users that they cannot control whether links will be created when calling the PrefabSystemComponentInterfac. I'm pretty sure that we need to add this control later because I can easily see systems needing to specify whether they want to create links or not. One such use case is the work being done for procedutral prefabs where they call the SystemComponentInterface directly. 

Rather than having these 2 flavors, I recommend just having one 'createPrefab' like before and give users the ability to customize it according to their needs by providing 'shouldCreateLinks', 'containerEntity' and 'parentInstance'. Since you have modified Prefab::Instance constructor to take different parameters, you can construct the instance with their specified options accordingly.